src/main/java/com/onevizion/scmdb/vo/SqlScript.java

```diff
@@ -177,6 +188,15 @@ public boolean equals(Object o) {
 
     @Override
     public int compareTo(SqlScript anotherScript) {
-        return name.compareTo(anotherScript.getName());
+        return orderNumber - anotherScript.getOrderNumber();
+    }
+
+    public static int extractOrderNumber(String scriptName) {
+        String[] parts = scriptName.split("_");
+        if (parts.length >= 1 && NumberUtils.isDigits(parts[0])) {
+            return Integer.parseInt(parts[0]);
+        } else {
+            return -1;
```

Comment:
throw exception if can't extract script number

Reply:
If we added throwing errors in the absence of a sequence number, then scmdb will crash if there are Dev Scripts in the folder.

Reply:
Yes, I forgot about it. Dev scripts - are scripts without numbers, previously used during development in branches. Now it can be deprecated, create an inquiry for the backend team and propose to drop this feature and always ignore scripts without numbers.
For now, you need to modify `compareTo()` to sort dep script with name instead of number

Reply:
Created Inquiry [94356](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100106051460&ttid=100091445)
Done.

Reply:
I'm not sure that now it works properly, anyway for such cases better to create `Comparator` like this
![image](https://user-images.githubusercontent.com/3282060/177345912-19b72a57-706f-403a-9612-4c2b49f88424.png)
For that, you need to change `orderNumber` to `Integer` and set it to `null` for dev scripts. Check is it works as expected or not.

Reply:
Done.